### PR TITLE
Improve font and font size selection.

### PR DESCRIPTION
**Pull request by jeff-h**
Use the system Font Panel.

### CODE REVIEW

**Comment by jeff-h** [comment]
There are two issues with this PR that I am finding hard to resolve:

* if you leave the Font Panel open when you quit the app, it reappears on launch.
* the Font Panel remains open when the system prefs window is closed. I am not sure if you're supposed to track this manually (store a `fontPanelIsOpen: Bool` and hide/show the panel manually) or whether there's some other way.

I think this PR partially raises the question in https://github.com/glushchenko/fsnotes/issues/20

File: FSNotes/PrefsViewController.swift [hunk 1]
```diff
@@ -83,16 +83,39 @@ class PrefsViewController: NSViewController {
...
+        controller?.editArea.font = UserDefaultsManagement.noteFont
+        setFontPreview()
+    }
+
+    func setFontPreview() {
+        fontPreview.font = UserDefaultsManagement.noteFont
+        fontPreview.stringValue = "\(UserDefaultsManagement.fontName) \(UserDefaultsManagement.fontSize)pt"
+    }
+    
+    override func validModesForFontPanel(_ fontPanel: NSFontPanel) -> Int {
```
Comment (glushchenko) [inline FSNotes/PrefsViewController.swift]: Xcode show error message here -> "Method does not override any method from its superclass". If comment out method, it works.

File: FSNotes/Base.lproj/Main.storyboard [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (glushchenko) [inline FSNotes/Base.lproj/Main.storyboard]: I think font size should be disabled here

Comment (jeff-h) [inline FSNotes/Base.lproj/Main.storyboard]: I am happy to do that — I agree that including font size here makes it practically impossible to cater for all fonts and sizes.

File: FSNotes/PrefsViewController.swift (hunk 1 above)
Comment (jeff-h) [inline FSNotes/PrefsViewController.swift]: Grr, I had a real painful time with that. I currently use the Xcode 9.0 beta which tells me that in 10.13 this method has changed (returns something different than `Int)`. I finally switched back to Xcode 8.3 and it suggested the above code, and required it to override.

Might have to leave it out for now, as I couldn't find any other way of doing it. It's not completely essential; it simply tells the Font Panel which parts to show. I was hiding the styling section at the top.

**Comment by glushchenko** [comment]
Crashed in preferences:

```
2017-08-13 12:22:12.756736+0300 FSNotes[58912:5704047] Unknown class _TtC7FSNotes21PrefsWindowController in Interface Builder file at path /Users/fluder/Library/Developer/Xcode/DerivedData/FSNotes-eclvndmlvhmmewaugwxqzcenvynz/Build/Products/Debug/FSNotes.app/Contents/Resources/Base.lproj/Main.storyboardc/NSWindowController-Ozi-at-KZg.nib.
2017-08-13 12:22:12.756813+0300 FSNotes[58912:5704047] Unknown class _TtC7FSNotes21PrefsWindowController in Interface Builder file at path /Users/fluder/Library/Developer/Xcode/DerivedData/FSNotes-eclvndmlvhmmewaugwxqzcenvynz/Build/Products/Debug/FSNotes.app/Contents/Resources/Base.lproj/Main.storyboardc/NSWindowController-Ozi-at-KZg.nib.
2017-08-13 12:22:12.756866+0300 FSNotes[58912:5704047] Unknown class _TtC7FSNotes21PrefsWindowController in Interface Builder file at path /Users/fluder/Library/Developer/Xcode/DerivedData/FSNotes-eclvndmlvhmmewaugwxqzcenvynz/Build/Products/Debug/FSNotes.app/Contents/Resources/Base.lproj/Main.storyboardc/NSWindowController-Ozi-at-KZg.nib.
2017-08-13 12:22:12.757239+0300 FSNotes[58912:5704047] Unknown class _TtC7FSNotes21PrefsWindowController in Interface Builder file at path /Users/fluder/Library/Developer/Xcode/DerivedData/FSNotes-eclvndmlvhmmewaugwxqzcenvynz/Build/Products/Debug/FSNotes.app/Contents/Resources/Base.lproj/Main.storyboardc/NSWindowController-Ozi-at-KZg.nib.
2017-08-13 12:22:12.757297+0300 FSNotes[58912:5704047] Unknown class _TtC7FSNotes21PrefsWindowController in Interface Builder file at path /Users/fluder/Library/Developer/Xcode/DerivedData/FSNotes-eclvndmlvhmmewaugwxqzcenvynz/Build/Products/Debug/FSNotes.app/Contents/Resources/Base.lproj/Main.storyboardc/NSWindowController-Ozi-at-KZg.nib.
2017-08-13 12:22:12.757324+0300 FSNotes[58912:5704047] Unknown class _TtC7FSNotes21PrefsWindowController in Interface Builder file at path /Users/fluder/Library/Developer/Xcode/DerivedData/FSNotes-eclvndmlvhmmewaugwxqzcenvynz/Build/Products/Debug/FSNotes.app/Contents/Resources/Base.lproj/Main.storyboardc/NSWindowController-Ozi-at-KZg.nib.
2017-08-13 12:22:12.757345+0300 FSNotes[58912:5704047] Unknown class _TtC7FSNotes21PrefsWindowController in Interface Builder file at path /Users/fluder/Library/Developer/Xcode/DerivedData/FSNotes-eclvndmlvhmmewaugwxqzcenvynz/Build/Products/Debug/FSNotes.app/Contents/Resources/Base.lproj/Main.storyboardc/NSWindowController-Ozi-at-KZg.nib.
2017-08-13 12:22:12.758802+0300 FSNotes[58912:5704047] Failed to connect (storyboard) outlet from ((null)) to (NSStoryboard): missing setter or instance variable
2017-08-13 12:22:12.761419+0300 FSNotes[58912:5704047] Failed to connect (window) outlet from ((null)) to (NSWindow): missing setter or instance variable
2017-08-13 12:22:12.766725+0300 FSNotes[58912:5704047] *** Assertion failure in -[NSStoryboard instantiateControllerWithIdentifier:], /BuildRoot/Library/Caches/com.apple.xbs/Sources/AppKit/AppKit-1548/AppKit.subproj/Storyboarding/NSStoryboard.m:304
2017-08-13 12:22:12.767391+0300 FSNotes[58912:5704047] [General] Could not load the scene view controller for identifier 'NSWindowController-Ozi-at-KZg'
2017-08-13 12:22:12.769253+0300 FSNotes[58912:5704047] [General] (
	0   CoreFoundation                      0x00007fff308675db __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x00007fff57546c76 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff3086d342 +[NSException raise:format:arguments:] + 98
	3   Foundation                          0x00007fff329a9500 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 193
	4   AppKit                              0x00007fff2e625be7 -[NSStoryboard instantiateControllerWithIdentifier:] + 396
	5   AppKit                              0x00007fff2e29a087 -[NSStoryboardSegueTemplate _perform:] + 70
	6   AppKit                              0x00007fff2e29b692 -[NSStoryboardShowSegueTemplate _perform:] + 275
	7   AppKit                              0x00007fff2e59c2e6 -[NSApplication(NSResponder) sendAction:to:from:] + 312
	8   AppKit                              0x00007fff2e033ed7 -[NSMenuItem _corePerformAction] + 323
	9   AppKit                              0x00007fff2e033c5f -[NSCarbonMenuImpl performActionWithHighlightingForItemAtIndex:] + 114
	10  AppKit                              0x00007fff2e032ac3 -[NSMenu performKeyEquivalent:] + 363
	11  AppKit                              0x00007fff2e59af4d routeKeyEquivalent + 884
	12  AppKit                              0x00007fff2e59849d -[NSApplication(NSEvent) sendEvent:] + 1041
	13  AppKit                              0x00007fff2ddfc9cc -[NSApplication run] + 812
	14  AppKit                              0x00007fff2ddcbcb2 NSApplicationMain + 804
	15  FSNotes                             0x00000001000195ed main + 13
	16  libdyld.dylib                       0x00007fff5813417d start + 1
	17  ???                                 0x0000000000000003 0x0 + 3
)
```

**Comment by jeff-h** [comment]
Is it possible Clean (shift-cmd-K) will fix that? It runs for me just fine.

**Comment by glushchenko** [comment]
I found solution, https://github.com/glushchenko/fsnotes/commit/2432652cc198ce046307937b0cb70acd234bbadf

**Pull request merged**